packages/dd-trace/src/appsec/stack_trace.js

```diff
@@ -27,19 +27,17 @@ function getCallSiteList (maxDepth = 100) {
 }
 
 function filterOutFramesFromLibrary (callSiteList) {
-  return callSiteList.filter(callSite => !callSite.getFileName().includes(ddBasePath))
+  return callSiteList.filter(callSite => !callSite.getFileName()?.startsWith(ddBasePath))
```

Comment:
adding that optional chaining has a side effect:
if startsWith is not a method of the intermediary value, that means it's probably not a string, so null or undefined ? if that's the case the `!` at the start of the arrow function will turn it into a true, and filter() will keep it in the array, i don't think that's intended right ? or what was your thought behind adding the optional chaining ?

Reply:
It's intended.

Reply:
And do we want to keep those in callSiteList ?